src/primitive/particle.rs

```diff
@@ -0,0 +1,272 @@
+//! Particle primitive
+
+use std::marker;
+
+use cgmath::{BaseFloat, Point2, Point3};
+use cgmath::prelude::*;
+
+use Ray;
+use prelude::*;
+
+/// Represents a particle in space.
+///
+/// Only have implementations of intersection tests for movement,
+/// using `(Particle, P, P)` where `P` is the positional `Point` of the particle, the first P is the
```

Comment:
let's use `(Particle<P>, Range<P>)` instead for better clarity?

Reply:
That's a good idea.